analytics.js

```diff
@@ -1271,6 +1284,12 @@ if (isValidArgsArray) {
     }
     i += 1;
   }
+
+  if (instance.toBeProcessedArray.length && instance.toBeProcessedArray[0][0] === defaultMethod) {
+    instance.toBeProcessedArray[0].shift();
+    instance[defaultMethod](...instance.toBeProcessedArray[0]);
```

Comment:
If we execute the `load` methods here. All other methods will be ignored in the normal case when no polyfill is loaded. Because the `instance.toBeProcessedArray` is currently empty 
I think we should double check the test cases too because I see all the tests were passed

Reply:
Yes. @MoumitaM, the polyfill-based load problem should be solved differently.
Let's revert this commit.

Reply:
Sorry. It's my bad. Actually, other methods won't be ignored because you make a manual call of `processDataInAnalyticsArray` after calling the `load` method. But it shows another error.
```js
// assume that instance.toBeProcessedArray = [['load', 'abc', 'xyz']]
instance.toBeProcessedArray[0].shift();
// now instance.toBeProcessedArray = [['abc', 'xyz']]
// executing the `load` method
instance[defaultMethod](...instance.toBeProcessedArray[0]);
// the function processDataInAnalyticsArray will be called inside the `load` method. Then it will throw an error since instance.toBeProcessedArray is equal to `[['abc', 'xyz']]`
// this line is executed after the function processDataInAnalyticsArray is called but it's too late
instance.toBeProcessedArray.shift();

```
Another thing is that the code flow is a bit complicated. The function `processDataInAnalyticsArray` is called at 3 different places. It's way hard to follow and maintain. My suggestion is: 
- Remove the `load` method from `instance.toBeProcessedArray` before executing it
- Push all other methods into `instance.toBeProcessedArray` before executing the `load` method
- Only call the `processDataInAnalyticsArray` function inside the `loadAfterPolyfill` function

This will ensure that all the methods will be executed after the `load` method regardless the polyfill is loaded or not.

Reply:
Addressed the concerns.